src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -271,18 +277,34 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
 
                                         break;
                                     }
+                                } else
+                                {
+                                    logDebugBuilder.AppendLine(LoggingConstants.RefreshKeyValueUnchanged + watchedKey);
                                 }
                             }
 
                             if (refreshAll)
                             {
                                 // Trigger a single load-all operation if a change was detected in one or more key-values with refreshAll: true
                                 applicationSettings = await LoadAll(client, cancellationToken).ConfigureAwait(false);
-
+                                logInfoBuilder.AppendLine(LoggingConstants.RefreshConfigurationUpdatedSuccess + endpoint);
+                                logDebugBuilder.AppendLine(LoggingConstants.RefreshConfigurationUpdatedSuccess + endpoint);
                                 return;
                             }
 
+                            if (keyValueChanges.Any())
+                            {
+                                logInfoBuilder.AppendLine(LoggingConstants.RefreshKeyValueUpdatedSuccess + endpoint);
+                                logDebugBuilder.AppendLine(LoggingConstants.RefreshKeyValueUpdatedSuccess + endpoint);

```

Comment:
We dont need to log success message again in the debug logger. We can just add the key name in debug logs, info logs will always contain the generic success message anyway.